experimental/rsocket-src/RSocketRequester.cpp

```diff
@@ -37,10 +37,40 @@ RSocketRequester::~RSocketRequester() {
   LOG(INFO) << "RSocketRequester => destroy";
 }
 
-std::shared_ptr<Subscriber<Payload>> RSocketRequester::requestChannel(
-    std::shared_ptr<Subscriber<Payload>> responseSink) {
-  // TODO need to runInEventBaseThread like other request methods
-  return reactiveSocket_->requestChannel(std::move(responseSink));
+yarpl::Reference<yarpl::flowable::Flowable<reactivesocket::Payload>>
+RSocketRequester::requestChannel(
+    yarpl::Reference<yarpl::flowable::Flowable<reactivesocket::Payload>>
+        payloads) {
+  auto& eb = eventBase_;
+  auto srs = reactiveSocket_;
+
+    LOG(INFO) << "requestChannel executing ";
+
+  return yarpl::flowable::Flowables::fromPublisher<Payload>([
+    &eb,
```

Comment:
Doing this along with `auto& eb = eventBase_;` makes me think you should just capture `this` by value and use `eventBase_` directly in the lambda.  It'll be less confusing I think.